File: 2016 maj/c++/3.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

string encode(string s, int k)
{
    string ans;
    for (char c: s)
        ans.push_back(char('A' + (c - 'A' + k) % 26));
    return ans;
}

bool valid(string s1, string s2)
{
    for (int k = 0; k < 26; k++)
        if (encode(s1, k) == s2)
            return true;
    return false;
}

int main()
{
    ifstream in("dane_6_3.txt");
    ofstream out("wyniki_6_3.txt");

    for (int i = 1; i <= 3000; i++)
    {
        string s1, s2;
        in >> s1 >> s2;
        if (!valid(s1, s2))
            out << s1 << '\n';
    }
}

File: 2016 maj/python/3.py
with open("dane_6_3.txt", "r") as f:
    file = f.read().strip().split('\n')

def encode(s, key = 107):
    return ''.join([chr(65 + (ord(i) - 65 + key) % 26) for i in s])

def valid(s1, s2):
    for key in range(26):
        if encode(s1, key) == s2:
            return True
    return False

with open("wyniki_6_3.txt", "w") as f:
    for s in file:
        s = s.split()
        if not valid(s[0], s[1]):
            f.write(s[0] + '\n')